### PR TITLE
Define generic boards for STM32L051K8U and -T

### PR DESCRIPTION
**Pull request by ademuri**
This PR adds generic board definitions for the STM32L051K8U and STM32L051K8T variants. They are generated using STM32Cube, following [the documentation in the wiki](https://github.com/stm32duino/Arduino_Core_STM32/wiki/Add-a-new-variant-(board)).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -617,6 +617,7 @@ User can add a STM32 based board following this [wiki](https://github.com/stm32d
 | :green_heart:  | STM32L041G6 | Generic Board | *2.4.0* |  |
 | :green_heart: | STM32L041K6T | Generic Board | *2.0.0* |  |
 | :green_heart: | STM32L051C6<br>STM32L051C8 | Generic Board | *2.0.0* |  |
+| :green_heart: | STM32L051K8 | Generic Board | *2.8.0* |  |
```
Comment (fpistm) [inline README.md:620]:
```suggestion
| :yellow_heart: | STM32L051K8 | Generic Board | **2.8.0** |  |
```

File: boards.txt [hunk 2]
```diff
@@ -8456,6 +8456,22 @@ GenL0.menu.pnum.GENERIC_L051C8UX.build.board=GENERIC_L051C8UX
 GenL0.menu.pnum.GENERIC_L051C8UX.build.product_line=STM32L051xx
 GenL0.menu.pnum.GENERIC_L051C8UX.build.variant=STM32L0xx/L051C(6-8)(T-U)
 
+# Generic L051K8Tx
```
Comment (fpistm) [inline boards.txt:8459]: The L051K6Tx should also be added.

File: boards.txt [hunk 3]
```diff
@@ -8456,6 +8456,22 @@ GenL0.menu.pnum.GENERIC_L051C8UX.build.board=GENERIC_L051C8UX
...
+GenL0.menu.pnum.GENERIC_L051K8TX=Generic L051K8Tx
+GenL0.menu.pnum.GENERIC_L051K8TX.upload.maximum_size=65536
+GenL0.menu.pnum.GENERIC_L051K8TX.upload.maximum_data_size=8192
+GenL0.menu.pnum.GENERIC_L051K8TX.build.board=GENERIC_L051K8TX
+GenL0.menu.pnum.GENERIC_L051K8TX.build.product_line=STM32L051xx
+GenL0.menu.pnum.GENERIC_L051K8TX.build.variant=STM32L0xx/L051K(6-8)T
+
+# Generic L051K8Ux
```
Comment (fpistm) [inline boards.txt:8467]: The L051K6Ux should also be added.

File: README.md (hunk 1 above)
Comment (fpistm) [inline README.md:620]: As well the  L051K6Tx and L051K6Ux should also be referenced.

**Review by fpistm — CHANGES_REQUESTED** [review]
Thanks for the updated PR.
Misses the L051K6Tx and L051K6Ux declaration.

File: README.md (hunk 1 above)
Comment (ademuri) [reply README.md]: Done.

File: boards.txt (hunk 2 above)
Comment (ademuri) [reply boards.txt:8468]: Done.

File: boards.txt (hunk 3 above)
Comment (ademuri) [reply boards.txt:8484]: Done.

File: README.md (hunk 1 above)
Comment (ademuri) [reply README.md]: Done.

**Comment by fpistm** [comment]
Thanks for the quick update.

File: README.md [hunk 4]
```diff
@@ -617,6 +617,10 @@ User can add a STM32 based board following this [wiki](https://github.com/stm32d
 | :green_heart:  | STM32L041G6 | Generic Board | *2.4.0* |  |
 | :green_heart: | STM32L041K6T | Generic Board | *2.0.0* |  |
 | :green_heart: | STM32L051C6<br>STM32L051C8 | Generic Board | *2.0.0* |  |
+| :yellow_heart: | STM32L051K6T | Generic Board | **2.8.0** |  |
+| :yellow_heart: | STM32L051K8T | Generic Board | **2.8.0** |  |
+| :yellow_heart: | STM32L051K6U | Generic Board | **2.8.0** |  |
+| :yellow_heart: | STM32L051K8U | Generic Board | **2.8.0** |  |
```
Comment (fpistm) [inline README.md:623]: Can be simplified as it references all version (only T and U).
```suggestion
| :yellow_heart: | STM32L051K6<br>STM32L051K8 | Generic Board | **2.8.0** |  |
```

File: boards.txt [hunk 5]
```diff
@@ -8456,6 +8456,39 @@ GenL0.menu.pnum.GENERIC_L051C8UX.build.board=GENERIC_L051C8UX
...
+GenL0.menu.pnum.GENERIC_L051K8TX.build.product_line=STM32L051xx
+GenL0.menu.pnum.GENERIC_L051K8TX.build.variant=STM32L0xx/L051K(6-8)T
+
+# Generic L051K6Ux
+GenL0.menu.pnum.GENERIC_L051K6UX=Generic L051K6Ux
+GenL0.menu.pnum.GENERIC_L051K6UX.upload.maximum_size=32768
+GenL0.menu.pnum.GENERIC_L051K6UX.upload.maximum_data_size=8192
+GenL0.menu.pnum.GENERIC_L051K6UX.build.board=GENERIC_L051K6UX
+GenL0.menu.pnum.GENERIC_L051K6UX.build.product_line=STM32L051xx
+GenL0.menu.pnum.GENERIC_L051K6UX.build.variant=STM32L0xx/L051K(6-8)U
```
Comment (fpistm) [inline boards.txt:8482]: Alphabetical order.
```suggestion
# Generic L051K6Ux
GenL0.menu.pnum.GENERIC_L051K6UX=Generic L051K6Ux
GenL0.menu.pnum.GENERIC_L051K6UX.upload.maximum_size=32768
GenL0.menu.pnum.GENERIC_L051K6UX.upload.maximum_data_size=8192
GenL0.menu.pnum.GENERIC_L051K6UX.build.board=GENERIC_L051K6UX
GenL0.menu.pnum.GENERIC_L051K6UX.build.product_line=STM32L051xx
GenL0.menu.pnum.GENERIC_L051K6UX.build.variant=STM32L0xx/L051K(6-8)U

# Generic L051K8Tx
GenL0.menu.pnum.GENERIC_L051K8TX=Generic L051K8Tx
GenL0.menu.pnum.GENERIC_L051K8TX.upload.maximum_size=65536
GenL0.menu.pnum.GENERIC_L051K8TX.upload.maximum_data_size=8192
GenL0.menu.pnum.GENERIC_L051K8TX.build.board=GENERIC_L051K8TX
GenL0.menu.pnum.GENERIC_L051K8TX.build.product_line=STM32L051xx
GenL0.menu.pnum.GENERIC_L051K8TX.build.variant=STM32L0xx/L051K(6-8)T
```

**Review by fpistm — CHANGES_REQUESTED** [review]
Thanks for the update.
Small changes then ready for merge. :+1:

File: README.md (hunk 4 above)
Comment (ademuri) [reply README.md]: Done.

File: boards.txt (hunk 5 above)
Comment (ademuri) [reply boards.txt]: Done.

**Review by fpistm — APPROVED** [review]
Thanks for your contribution.
LGTM.

**Pull request merged**